<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to msigdbr • msigdbr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to msigdbr">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">msigdbr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">10.0.2.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<ul class="navbar-nav">
<li class="active nav-item"><a class="nav-link" href="../articles/msigdbr-intro.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/igordot/msigdbr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to msigdbr</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/igordot/msigdbr/blob/main/vignettes/articles/msigdbr-intro.Rmd" class="external-link"><code>vignettes/articles/msigdbr-intro.Rmd</code></a></small>
      <div class="d-none name"><code>msigdbr-intro.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Pathway analysis is a common task in genomics research and there are
many available R-based software tools. Depending on the tool, it may be
necessary to import the pathways, translate genes to the appropriate
species, convert between symbols and IDs, and format the resulting
object.</p>
<p>The msigdbr R package provides Molecular Signatures Database (MSigDB)
gene sets typically used with the Gene Set Enrichment Analysis (GSEA)
software:</p>
<ul>
<li>in an R-friendly “<a href="https://r4ds.hadley.nz/data-tidy.html" class="external-link">tidy</a>” format with one
gene pair per row</li>
<li>for multiple frequently studied model organisms, such as mouse, rat,
pig, zebrafish, fly, and yeast, in addition to the original human
genes</li>
<li>as gene symbols as well as NCBI Entrez and Ensembl IDs</li>
<li>without accessing external resources requiring an active internet
connection</li>
</ul>
<p>Please be aware that the orthologs were computationally predicted at
the gene level. The full pathways may not be well conserved across
species.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The package can be installed from <a href="https://cran.r-project.org/package=msigdbr" class="external-link">CRAN</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"msigdbr"</span><span class="op">)</span></span></code></pre></div>
<p>The package includes only a small subset of the full MSigDB database
due to CRAN size limitations. Please install the msigdbdf package to
access the full MSigDB database:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"msigdbdf"</span>, repos <span class="op">=</span> <span class="st">"https://igordot.r-universe.dev"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Load package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/">msigdbr</a></span><span class="op">)</span></span></code></pre></div>
<p>All gene sets in the database can be retrieved by specifying a
species of interest.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">all_gene_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt;   gene_symbol ncbi_gene       ensembl_gene db_gene_symbol db_ncbi_gene</span></span>
<span><span class="co">#&gt; 1       Abcc4    239273 ENSMUSG00000032849          ABCC4        10257</span></span>
<span><span class="co">#&gt; 2    Abraxas2    109359 ENSMUSG00000030965       ABRAXAS2        23172</span></span>
<span><span class="co">#&gt; 3       Actn4     60595 ENSMUSG00000054808          ACTN4           81</span></span>
<span><span class="co">#&gt; 4       Acvr1     11477 ENSMUSG00000026836          ACVR1           90</span></span>
<span><span class="co">#&gt; 5       Adam9     11502 ENSMUSG00000031555          ADAM9         8754</span></span>
<span><span class="co">#&gt; 6     Adamts5     23794 ENSMUSG00000022894        ADAMTS5        11096</span></span>
<span><span class="co">#&gt;   db_ensembl_gene source_gene  gs_id        gs_name gs_collection</span></span>
<span><span class="co">#&gt; 1 ENSG00000125257       ABCC4 M12609 AAACCAC_MIR140            C3</span></span>
<span><span class="co">#&gt; 2 ENSG00000165660    KIAA0157 M12609 AAACCAC_MIR140            C3</span></span>
<span><span class="co">#&gt; 3 ENSG00000130402       ACTN4 M12609 AAACCAC_MIR140            C3</span></span>
<span><span class="co">#&gt; 4 ENSG00000115170       ACVR1 M12609 AAACCAC_MIR140            C3</span></span>
<span><span class="co">#&gt; 5 ENSG00000168615       ADAM9 M12609 AAACCAC_MIR140            C3</span></span>
<span><span class="co">#&gt; 6 ENSG00000154736     ADAMTS5 M12609 AAACCAC_MIR140            C3</span></span>
<span><span class="co">#&gt;   gs_subcollection gs_collection_name</span></span>
<span><span class="co">#&gt; 1   MIR:MIR_LEGACY         MIR_Legacy</span></span>
<span><span class="co">#&gt; 2   MIR:MIR_LEGACY         MIR_Legacy</span></span>
<span><span class="co">#&gt; 3   MIR:MIR_LEGACY         MIR_Legacy</span></span>
<span><span class="co">#&gt; 4   MIR:MIR_LEGACY         MIR_Legacy</span></span>
<span><span class="co">#&gt; 5   MIR:MIR_LEGACY         MIR_Legacy</span></span>
<span><span class="co">#&gt; 6   MIR:MIR_LEGACY         MIR_Legacy</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                           gs_description</span></span>
<span><span class="co">#&gt; 1 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</span></span>
<span><span class="co">#&gt; 2 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</span></span>
<span><span class="co">#&gt; 3 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</span></span>
<span><span class="co">#&gt; 4 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</span></span>
<span><span class="co">#&gt; 5 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</span></span>
<span><span class="co">#&gt; 6 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</span></span>
<span><span class="co">#&gt;   gs_source_species gs_pmid gs_geoid gs_exact_source gs_url db_version</span></span>
<span><span class="co">#&gt; 1                HS                                          2024.1.Hs</span></span>
<span><span class="co">#&gt; 2                HS                                          2024.1.Hs</span></span>
<span><span class="co">#&gt; 3                HS                                          2024.1.Hs</span></span>
<span><span class="co">#&gt; 4                HS                                          2024.1.Hs</span></span>
<span><span class="co">#&gt; 5                HS                                          2024.1.Hs</span></span>
<span><span class="co">#&gt; 6                HS                                          2024.1.Hs</span></span>
<span><span class="co">#&gt;   db_target_species ortholog_taxon_id</span></span>
<span><span class="co">#&gt; 1                HS             10090</span></span>
<span><span class="co">#&gt; 2                HS             10090</span></span>
<span><span class="co">#&gt; 3                HS             10090</span></span>
<span><span class="co">#&gt; 4                HS             10090</span></span>
<span><span class="co">#&gt; 5                HS             10090</span></span>
<span><span class="co">#&gt; 6                HS             10090</span></span>
<span><span class="co">#&gt;                                                                                ortholog_sources</span></span>
<span><span class="co">#&gt; 1                          Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther</span></span>
<span><span class="co">#&gt; 2 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 3 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 4 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 5           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt; 6           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt;   num_ortholog_sources</span></span>
<span><span class="co">#&gt; 1                    9</span></span>
<span><span class="co">#&gt; 2                   12</span></span>
<span><span class="co">#&gt; 3                   12</span></span>
<span><span class="co">#&gt; 4                   12</span></span>
<span><span class="co">#&gt; 5                   11</span></span>
<span><span class="co">#&gt; 6                   11</span></span></code></pre></div>
<p>The <code>gene_symbol</code>, <code>ncbi_gene</code>, and
<code>ensembl_gene</code> columns are the gene identifiers for the
requested species. The <code>db_gene_symbol</code>,
<code>db_ncbi_gene</code>, <code>db_ensembl_gene</code> columns are the
gene identifiers in the database. The <code>source_gene</code> column
are the gene identifiers in the original publication. The
<code>gs_*</code> columns provide details about the gene sets. The
<code>db_*</code> columns provide details about the MSigDB database, so
should be identical for the entire data frame.</p>
<p>You can retrieve data just for a specific collection, such as the
Hallmark gene sets.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, collection <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">h_gene_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt;   gene_symbol ncbi_gene       ensembl_gene db_gene_symbol db_ncbi_gene</span></span>
<span><span class="co">#&gt; 1       Abca1     11303 ENSMUSG00000015243          ABCA1           19</span></span>
<span><span class="co">#&gt; 2       Abcb8     74610 ENSMUSG00000028973          ABCB8        11194</span></span>
<span><span class="co">#&gt; 3       Acaa2     52538 ENSMUSG00000036880          ACAA2        10449</span></span>
<span><span class="co">#&gt; 4       Acadl     11363 ENSMUSG00000026003          ACADL           33</span></span>
<span><span class="co">#&gt; 5       Acadm     11364 ENSMUSG00000062908          ACADM           34</span></span>
<span><span class="co">#&gt; 6       Acads     11409 ENSMUSG00000029545          ACADS           35</span></span>
<span><span class="co">#&gt;   db_ensembl_gene source_gene gs_id               gs_name gs_collection</span></span>
<span><span class="co">#&gt; 1 ENSG00000165029       ABCA1 M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 2 ENSG00000197150       ABCB8 M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 3 ENSG00000167315       ACAA2 M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 4 ENSG00000115361       ACADL M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 5 ENSG00000117054       ACADM M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 6 ENSG00000122971       ACADS M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt;   gs_subcollection gs_collection_name</span></span>
<span><span class="co">#&gt; 1                            Hallmark</span></span>
<span><span class="co">#&gt; 2                            Hallmark</span></span>
<span><span class="co">#&gt; 3                            Hallmark</span></span>
<span><span class="co">#&gt; 4                            Hallmark</span></span>
<span><span class="co">#&gt; 5                            Hallmark</span></span>
<span><span class="co">#&gt; 6                            Hallmark</span></span>
<span><span class="co">#&gt;                                                        gs_description</span></span>
<span><span class="co">#&gt; 1 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 2 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 3 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 4 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 5 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 6 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt;   gs_source_species  gs_pmid gs_geoid gs_exact_source gs_url db_version</span></span>
<span><span class="co">#&gt; 1                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 2                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 3                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 4                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 5                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 6                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt;   db_target_species ortholog_taxon_id</span></span>
<span><span class="co">#&gt; 1                HS             10090</span></span>
<span><span class="co">#&gt; 2                HS             10090</span></span>
<span><span class="co">#&gt; 3                HS             10090</span></span>
<span><span class="co">#&gt; 4                HS             10090</span></span>
<span><span class="co">#&gt; 5                HS             10090</span></span>
<span><span class="co">#&gt; 6                HS             10090</span></span>
<span><span class="co">#&gt;                                                                                ortholog_sources</span></span>
<span><span class="co">#&gt; 1           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt; 2 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 3           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt; 4 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 5 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 6           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt;   num_ortholog_sources</span></span>
<span><span class="co">#&gt; 1                   11</span></span>
<span><span class="co">#&gt; 2                   12</span></span>
<span><span class="co">#&gt; 3                   11</span></span>
<span><span class="co">#&gt; 4                   12</span></span>
<span><span class="co">#&gt; 5                   12</span></span>
<span><span class="co">#&gt; 6                   11</span></span></code></pre></div>
<p>You can specify a sub-collection, such as C2 (curated) CGP (chemical
and genetic perturbations) gene sets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cgp_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, collection <span class="op">=</span> <span class="st">"C2"</span>, subcollection <span class="op">=</span> <span class="st">"CGP"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cgp_gene_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt;   gene_symbol ncbi_gene       ensembl_gene db_gene_symbol db_ncbi_gene</span></span>
<span><span class="co">#&gt; 1       Ahnak     66395 ENSMUSG00000069833          AHNAK        79026</span></span>
<span><span class="co">#&gt; 2       Alcam     11658 ENSMUSG00000022636          ALCAM          214</span></span>
<span><span class="co">#&gt; 3     Ankrd40     71452 ENSMUSG00000020864        ANKRD40        91369</span></span>
<span><span class="co">#&gt; 4      Arid1a     93760 ENSMUSG00000007880         ARID1A         8289</span></span>
<span><span class="co">#&gt; 5      Bckdhb     12040 ENSMUSG00000032263         BCKDHB          594</span></span>
<span><span class="co">#&gt; 6    AU021092    239691 ENSMUSG00000051669       C16orf89       146556</span></span>
<span><span class="co">#&gt;   db_ensembl_gene source_gene gs_id                  gs_name gs_collection</span></span>
<span><span class="co">#&gt; 1 ENSG00000124942    AA657044 M1423 ABBUD_LIF_SIGNALING_1_DN            C2</span></span>
<span><span class="co">#&gt; 2 ENSG00000170017      L25274 M1423 ABBUD_LIF_SIGNALING_1_DN            C2</span></span>
<span><span class="co">#&gt; 3 ENSG00000154945    AW125433 M1423 ABBUD_LIF_SIGNALING_1_DN            C2</span></span>
<span><span class="co">#&gt; 4 ENSG00000117713    AI842326 M1423 ABBUD_LIF_SIGNALING_1_DN            C2</span></span>
<span><span class="co">#&gt; 5 ENSG00000083123      L16992 M1423 ABBUD_LIF_SIGNALING_1_DN            C2</span></span>
<span><span class="co">#&gt; 6 ENSG00000153446    AW045808 M1423 ABBUD_LIF_SIGNALING_1_DN            C2</span></span>
<span><span class="co">#&gt;   gs_subcollection                 gs_collection_name</span></span>
<span><span class="co">#&gt; 1              CGP Chemical and Genetic Perturbations</span></span>
<span><span class="co">#&gt; 2              CGP Chemical and Genetic Perturbations</span></span>
<span><span class="co">#&gt; 3              CGP Chemical and Genetic Perturbations</span></span>
<span><span class="co">#&gt; 4              CGP Chemical and Genetic Perturbations</span></span>
<span><span class="co">#&gt; 5              CGP Chemical and Genetic Perturbations</span></span>
<span><span class="co">#&gt; 6              CGP Chemical and Genetic Perturbations</span></span>
<span><span class="co">#&gt;                                                                                    gs_description</span></span>
<span><span class="co">#&gt; 1 Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].</span></span>
<span><span class="co">#&gt; 2 Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].</span></span>
<span><span class="co">#&gt; 3 Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].</span></span>
<span><span class="co">#&gt; 4 Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].</span></span>
<span><span class="co">#&gt; 5 Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].</span></span>
<span><span class="co">#&gt; 6 Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].</span></span>
<span><span class="co">#&gt;   gs_source_species  gs_pmid gs_geoid gs_exact_source gs_url db_version</span></span>
<span><span class="co">#&gt; 1                MM 14576184                  Table 2         2024.1.Hs</span></span>
<span><span class="co">#&gt; 2                MM 14576184                  Table 2         2024.1.Hs</span></span>
<span><span class="co">#&gt; 3                MM 14576184                  Table 2         2024.1.Hs</span></span>
<span><span class="co">#&gt; 4                MM 14576184                  Table 2         2024.1.Hs</span></span>
<span><span class="co">#&gt; 5                MM 14576184                  Table 2         2024.1.Hs</span></span>
<span><span class="co">#&gt; 6                MM 14576184                  Table 2         2024.1.Hs</span></span>
<span><span class="co">#&gt;   db_target_species ortholog_taxon_id</span></span>
<span><span class="co">#&gt; 1                HS             10090</span></span>
<span><span class="co">#&gt; 2                HS             10090</span></span>
<span><span class="co">#&gt; 3                HS             10090</span></span>
<span><span class="co">#&gt; 4                HS             10090</span></span>
<span><span class="co">#&gt; 5                HS             10090</span></span>
<span><span class="co">#&gt; 6                HS             10090</span></span>
<span><span class="co">#&gt;                                                                                ortholog_sources</span></span>
<span><span class="co">#&gt; 1           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt; 2 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 3 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 4 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 5 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 6 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt;   num_ortholog_sources</span></span>
<span><span class="co">#&gt; 1                   11</span></span>
<span><span class="co">#&gt; 2                   12</span></span>
<span><span class="co">#&gt; 3                   12</span></span>
<span><span class="co">#&gt; 4                   12</span></span>
<span><span class="co">#&gt; 5                   12</span></span>
<span><span class="co">#&gt; 6                   12</span></span></code></pre></div>
<p>If you require more precise filtering, the <code><a href="../reference/msigdbr.html">msigdbr()</a></code>
function output is a data frame that can be manipulated using standard
methods. For example, you can subset to a specific collection using
dplyr.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_gene_sets</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gs_collection</span> <span class="op">==</span> <span class="st">"H"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   gene_symbol ncbi_gene       ensembl_gene db_gene_symbol db_ncbi_gene</span></span>
<span><span class="co">#&gt; 1       Abca1     11303 ENSMUSG00000015243          ABCA1           19</span></span>
<span><span class="co">#&gt; 2       Abcb8     74610 ENSMUSG00000028973          ABCB8        11194</span></span>
<span><span class="co">#&gt; 3       Acaa2     52538 ENSMUSG00000036880          ACAA2        10449</span></span>
<span><span class="co">#&gt; 4       Acadl     11363 ENSMUSG00000026003          ACADL           33</span></span>
<span><span class="co">#&gt; 5       Acadm     11364 ENSMUSG00000062908          ACADM           34</span></span>
<span><span class="co">#&gt; 6       Acads     11409 ENSMUSG00000029545          ACADS           35</span></span>
<span><span class="co">#&gt;   db_ensembl_gene source_gene gs_id               gs_name gs_collection</span></span>
<span><span class="co">#&gt; 1 ENSG00000165029       ABCA1 M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 2 ENSG00000197150       ABCB8 M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 3 ENSG00000167315       ACAA2 M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 4 ENSG00000115361       ACADL M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 5 ENSG00000117054       ACADM M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt; 6 ENSG00000122971       ACADS M5905 HALLMARK_ADIPOGENESIS             H</span></span>
<span><span class="co">#&gt;   gs_subcollection gs_collection_name</span></span>
<span><span class="co">#&gt; 1                            Hallmark</span></span>
<span><span class="co">#&gt; 2                            Hallmark</span></span>
<span><span class="co">#&gt; 3                            Hallmark</span></span>
<span><span class="co">#&gt; 4                            Hallmark</span></span>
<span><span class="co">#&gt; 5                            Hallmark</span></span>
<span><span class="co">#&gt; 6                            Hallmark</span></span>
<span><span class="co">#&gt;                                                        gs_description</span></span>
<span><span class="co">#&gt; 1 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 2 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 3 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 4 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 5 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt; 6 Genes up-regulated during adipocyte differentiation (adipogenesis).</span></span>
<span><span class="co">#&gt;   gs_source_species  gs_pmid gs_geoid gs_exact_source gs_url db_version</span></span>
<span><span class="co">#&gt; 1                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 2                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 3                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 4                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 5                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt; 6                HS 26771021                                  2024.1.Hs</span></span>
<span><span class="co">#&gt;   db_target_species ortholog_taxon_id</span></span>
<span><span class="co">#&gt; 1                HS             10090</span></span>
<span><span class="co">#&gt; 2                HS             10090</span></span>
<span><span class="co">#&gt; 3                HS             10090</span></span>
<span><span class="co">#&gt; 4                HS             10090</span></span>
<span><span class="co">#&gt; 5                HS             10090</span></span>
<span><span class="co">#&gt; 6                HS             10090</span></span>
<span><span class="co">#&gt;                                                                                ortholog_sources</span></span>
<span><span class="co">#&gt; 1           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt; 2 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 3           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt; 4 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 5 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span></span>
<span><span class="co">#&gt; 6           EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span></span>
<span><span class="co">#&gt;   num_ortholog_sources</span></span>
<span><span class="co">#&gt; 1                   11</span></span>
<span><span class="co">#&gt; 2                   12</span></span>
<span><span class="co">#&gt; 3                   11</span></span>
<span><span class="co">#&gt; 4                   12</span></span>
<span><span class="co">#&gt; 5                   12</span></span>
<span><span class="co">#&gt; 6                   11</span></span></code></pre></div>
<p>The version of the MSigDB database is stored in the
<code>db_version</code> column of the returned data frame.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">all_gene_sets</span><span class="op">$</span><span class="va">db_version</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "2024.1.Hs"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="helper-functions">Helper functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a>
</h2>
<p>There are helper functions to assist with setting the
<code><a href="../reference/msigdbr.html">msigdbr()</a></code> parameters.</p>
<p>Use <code><a href="../reference/msigdbr_species.html">msigdbr_species()</a></code> to check the available species.
Both scientific and common names are acceptable for the
<code><a href="../reference/msigdbr.html">msigdbr()</a></code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/msigdbr_species.html">msigdbr_species</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 20 × 2</span></span></span>
<span><span class="co">#&gt;    species_name                    species_common_name                          </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Anolis carolinensis             Carolina anole, green anole                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Bos taurus                      bovine, cattle, cow, dairy cow, domestic cat…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Caenorhabditis elegans          <span style="color: #BB0000;">NA</span>                                           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Canis lupus familiaris          dog, dogs                                    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Danio rerio                     leopard danio, zebra danio, zebra fish, zebr…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Drosophila melanogaster         fruit fly                                    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Equus caballus                  domestic horse, equine, horse                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Felis catus                     cat, cats, domestic cat                      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gallus gallus                   bantam, chicken, chickens, Gallus domesticus </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Homo sapiens                    human                                        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Macaca mulatta                  rhesus macaque, rhesus macaques, Rhesus monk…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Monodelphis domestica           gray short-tailed opossum                    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> Mus musculus                    house mouse, mouse                           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> Ornithorhynchus anatinus        duck-billed platypus, duckbill platypus, pla…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> Pan troglodytes                 chimpanzee                                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> Rattus norvegicus               brown rat, Norway rat, rat, rats             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> Saccharomyces cerevisiae        baker's yeast, brewer's yeast, S. cerevisiae </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> Schizosaccharomyces pombe 972h- <span style="color: #BB0000;">NA</span>                                           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> Sus scrofa                      pig, pigs, swine, wild boar                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> Xenopus tropicalis              tropical clawed frog, western clawed frog</span></span></code></pre></div>
<p>Use <code><a href="../reference/msigdbr_collections.html">msigdbr_collections()</a></code> to check the available
collections.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/msigdbr_collections.html">msigdbr_collections</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;    gs_collection gs_subcollection                   gs_collection_name</span></span>
<span><span class="co">#&gt; 1             C1                                            Positional</span></span>
<span><span class="co">#&gt; 2             C2              CGP   Chemical and Genetic Perturbations</span></span>
<span><span class="co">#&gt; 3             C2               CP                   Canonical Pathways</span></span>
<span><span class="co">#&gt; 4             C2      CP:BIOCARTA                    BioCarta Pathways</span></span>
<span><span class="co">#&gt; 5             C2   CP:KEGG_LEGACY                 KEGG Legacy Pathways</span></span>
<span><span class="co">#&gt; 6             C2  CP:KEGG_MEDICUS                KEGG Medicus Pathways</span></span>
<span><span class="co">#&gt; 7             C2           CP:PID                         PID Pathways</span></span>
<span><span class="co">#&gt; 8             C2      CP:REACTOME                    Reactome Pathways</span></span>
<span><span class="co">#&gt; 9             C2  CP:WIKIPATHWAYS                         WikiPathways</span></span>
<span><span class="co">#&gt; 10            C3        MIR:MIRDB                                miRDB</span></span>
<span><span class="co">#&gt; 11            C3   MIR:MIR_LEGACY                           MIR_Legacy</span></span>
<span><span class="co">#&gt; 12            C3         TFT:GTRD                                 GTRD</span></span>
<span><span class="co">#&gt; 13            C3   TFT:TFT_LEGACY                           TFT_Legacy</span></span>
<span><span class="co">#&gt; 14            C4              3CA Curated Cancer Cell Atlas gene sets </span></span>
<span><span class="co">#&gt; 15            C4              CGN            Cancer Gene Neighborhoods</span></span>
<span><span class="co">#&gt; 16            C4               CM                       Cancer Modules</span></span>
<span><span class="co">#&gt; 17            C5            GO:BP                GO Biological Process</span></span>
<span><span class="co">#&gt; 18            C5            GO:CC                GO Cellular Component</span></span>
<span><span class="co">#&gt; 19            C5            GO:MF                GO Molecular Function</span></span>
<span><span class="co">#&gt; 20            C5              HPO             Human Phenotype Ontology</span></span>
<span><span class="co">#&gt; 21            C6                                   Oncogenic Signature</span></span>
<span><span class="co">#&gt; 22            C7      IMMUNESIGDB                          ImmuneSigDB</span></span>
<span><span class="co">#&gt; 23            C7              VAX                HIPC Vaccine Response</span></span>
<span><span class="co">#&gt; 24            C8                                   Cell Type Signature</span></span>
<span><span class="co">#&gt; 25             H                                              Hallmark</span></span>
<span><span class="co">#&gt;    num_genesets</span></span>
<span><span class="co">#&gt; 1           302</span></span>
<span><span class="co">#&gt; 2          3494</span></span>
<span><span class="co">#&gt; 3            19</span></span>
<span><span class="co">#&gt; 4           292</span></span>
<span><span class="co">#&gt; 5           186</span></span>
<span><span class="co">#&gt; 6           658</span></span>
<span><span class="co">#&gt; 7           196</span></span>
<span><span class="co">#&gt; 8          1736</span></span>
<span><span class="co">#&gt; 9           830</span></span>
<span><span class="co">#&gt; 10         2377</span></span>
<span><span class="co">#&gt; 11          221</span></span>
<span><span class="co">#&gt; 12          505</span></span>
<span><span class="co">#&gt; 13          610</span></span>
<span><span class="co">#&gt; 14          148</span></span>
<span><span class="co">#&gt; 15          427</span></span>
<span><span class="co">#&gt; 16          431</span></span>
<span><span class="co">#&gt; 17         7608</span></span>
<span><span class="co">#&gt; 18         1026</span></span>
<span><span class="co">#&gt; 19         1820</span></span>
<span><span class="co">#&gt; 20         5653</span></span>
<span><span class="co">#&gt; 21          189</span></span>
<span><span class="co">#&gt; 22         4872</span></span>
<span><span class="co">#&gt; 23          347</span></span>
<span><span class="co">#&gt; 24          840</span></span>
<span><span class="co">#&gt; 25           50</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pathway-enrichment-analysis">Pathway enrichment analysis<a class="anchor" aria-label="anchor" href="#pathway-enrichment-analysis"></a>
</h2>
<p>The msigdbr output can be used with various pathway analysis
packages.</p>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/clusterProfiler/" class="external-link">clusterProfiler</a>
with genes as NCBI/Entrez IDs.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_t2g</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">msigdbr_df</span>, <span class="va">gs_name</span>, <span class="va">ncbi_gene</span><span class="op">)</span></span>
<span><span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_ids_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/clusterProfiler/" class="external-link">clusterProfiler</a>
with genes as gene symbols.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_t2g</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">msigdbr_df</span>, <span class="va">gs_name</span>, <span class="va">gene_symbol</span><span class="op">)</span></span>
<span><span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_symbols_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/fgsea/" class="external-link">fgsea</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="fu">fgsea</span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/GSVA/" class="external-link">GSVA</a>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="fu">gsva</span><span class="op">(</span>gset.idx.list <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="potential-questions-and-concerns">Potential questions and concerns<a class="anchor" aria-label="anchor" href="#potential-questions-and-concerns"></a>
</h2>
<p><strong>Which version of MSigDB was used?</strong></p>
<p>The MSigDB version is stored in the <code>db_version</code> column of
the returned data frame. You can check the version used with
<code>unique(msigdbr_df$db_version)</code>.</p>
<p><strong>Why use this package when I can download the gene sets
directly from MSigDB?</strong></p>
<p>This package makes it more convenient to work with MSigDB gene sets
in R. You don’t need to download the GMT files and import them. You
don’t need to learn how to restructure the output to make it compatible
with downstream tools. You don’t need to convert the genes to your
organism if you are working with non-human data.</p>
<p><strong>Can I convert between human and mouse genes just by adjusting
gene capitalization?</strong></p>
<p>That will work for most, but not all, genes.</p>
<p><strong>Can I convert human genes to any organism myself instead of
using this package?</strong></p>
<p>One popular method is using the biomaRt package. You may still end up
with dozens of homologs for some genes, so additional cleanup may be
helpful.</p>
<p><strong>Aren’t there already other similar tools?</strong></p>
<p>There are a few resources that provide some of the msigdbr
functionality and served as an inspiration for this package. <a href="https://bioinf.wehi.edu.au/software/MSigDB/" class="external-link">WEHI</a> provides
MSigDB gene sets in R format for human and mouse. <a href="https://github.com/stephenturner/msigdf" class="external-link">MSigDF</a> and a more
recent <a href="https://github.com/ToledoEM/msigdf" class="external-link">ToledoEM/msigdf</a>
fork provide a tidyverse-friendly data frame. These are updated at
varying frequencies and may not be based on the latest version of
MSigDB. Since 2022, the GSEA/MSigDB team provides <a href="https://www.gsea-msigdb.org/gsea/msigdb/mouse/collections.jsp" class="external-link">collections
that are natively mouse</a> and don’t require orthology conversion.</p>
<p><strong>What if I have other questions?</strong></p>
<p>You can submit feedback and report bugs on <a href="https://github.com/igordot/msigdbr/issues" class="external-link">GitHub</a>.</p>
</div>
<div class="section level2">
<h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a>
</h2>
<p>The Molecular Signatures Database (MSigDB) is a collection of gene
sets originally created for use with the Gene Set Enrichment Analysis
(GSEA) software. To cite use of the underlying MSigDB data, reference
Subramanian, Tamayo, et al. (2005, PNAS) and one or more of the
following as appropriate: Liberzon, et al. (2011, Bioinformatics),
Liberzon, et al. (2015, Cell Systems), Castanza, et al. (2023, Nature
Methods) and also the source for the gene set.</p>
<p>Gene homologs are provided by HUGO Gene Nomenclature Committee at the
European Bioinformatics Institute which integrates the orthology
assertions predicted for human genes by eggNOG, Ensembl Compara, HGNC,
HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL,
Panther, PhylomeDB, TreeFam and ZFIN. For each human equivalent within
each species, only the ortholog supported by the largest number of
databases is used.</p>
<p>For information on how to cite the msigdbr R package, run
<code>citation("msigdbr")</code>.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Igor Dolgalev.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>



  <script data-goatcounter="https://igordot.goatcounter.com/count" async src="//gc.zgo.at/count.js"></script>
</body>
</html>
