<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to msigdbr • msigdbr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to msigdbr">
<meta property="og:description" content="msigdbr">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:creator" content="@bioigor">
<meta name="twitter:site" content="@bioigor">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-130263433-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-130263433-2');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">msigdbr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">7.5.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"></ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/msigdbr-intro.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="https://github.com/igordot/msigdbr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to msigdbr</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/igordot/msigdbr/blob/HEAD/vignettes/msigdbr-intro.Rmd" class="external-link"><code>vignettes/msigdbr-intro.Rmd</code></a></small>
      <div class="hidden name"><code>msigdbr-intro.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Pathway analysis is a common task in genomics research and there are many available R-based software tools. Depending on the tool, it may be necessary to import the pathways, translate genes to the appropriate species, convert between symbols and IDs, and format the resulting object.</p>
<p>The <code>msigdbr</code> R package provides Molecular Signatures Database (MSigDB) gene sets typically used with the Gene Set Enrichment Analysis (GSEA) software:</p>
<ul>
<li>in an R-friendly tidy/long format with one gene per row</li>
<li>for multiple frequently studied model organisms, such as mouse, rat, pig, zebrafish, fly, and yeast, in addition to the original human genes</li>
<li>as gene symbols as well as NCBI Entrez and Ensembl IDs</li>
<li>that can be installed and loaded as a package without requiring additional external files</li>
</ul>
<p>Please be aware that the homologs were computationally predicted for distinct genes. The full pathways may not be well conserved across species.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The package can be installed from <a href="https://cran.r-project.org/package=msigdbr" class="external-link">CRAN</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"msigdbr"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Load package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/">msigdbr</a></span><span class="op">)</span></span></code></pre></div>
<p>All gene sets in the database can be retrieved without specifying a collection/category.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">all_gene_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 18</span></span></span>
<span><span class="co">#&gt;   gs_cat gs_subcat  gs_name gene_symbol entrez_gene ensembl_gene human_gene…¹ human_entr…²</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> C3     MIR:MIR_L… AAACCA… Abcc4            <span style="text-decoration: underline;">239</span>273 ENSMUSG0000… ABCC4               <span style="text-decoration: underline;">10</span>257</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> C3     MIR:MIR_L… AAACCA… Abraxas2         <span style="text-decoration: underline;">109</span>359 ENSMUSG0000… ABRAXAS2            <span style="text-decoration: underline;">23</span>172</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> C3     MIR:MIR_L… AAACCA… Actn4             <span style="text-decoration: underline;">60</span>595 ENSMUSG0000… ACTN4                  81</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> C3     MIR:MIR_L… AAACCA… Acvr1             <span style="text-decoration: underline;">11</span>477 ENSMUSG0000… ACVR1                  90</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> C3     MIR:MIR_L… AAACCA… Adam9             <span style="text-decoration: underline;">11</span>502 ENSMUSG0000… ADAM9                <span style="text-decoration: underline;">8</span>754</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> C3     MIR:MIR_L… AAACCA… Adamts5           <span style="text-decoration: underline;">23</span>794 ENSMUSG0000… ADAMTS5             <span style="text-decoration: underline;">11</span>096</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 10 more variables: human_ensembl_gene &lt;chr&gt;, gs_id &lt;chr&gt;, gs_pmid &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;, gs_description &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   taxon_id &lt;int&gt;, ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;, and abbreviated</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variable names ¹​human_gene_symbol, ²​human_entrez_gene</span></span></span></code></pre></div>
<p>There is a helper function to show the available species. Either scientific or common names are acceptable.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/msigdbr_species.html">msigdbr_species</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 20 × 2</span></span></span>
<span><span class="co">#&gt;    species_name                    species_common_name                                    </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Anolis carolinensis             Carolina anole, green anole                            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Bos taurus                      bovine, cattle, cow, dairy cow, domestic cattle, domes…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Caenorhabditis elegans          <span style="color: #BB0000;">NA</span>                                                     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Canis lupus familiaris          dog, dogs                                              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Danio rerio                     leopard danio, zebra danio, zebra fish, zebrafish      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Drosophila melanogaster         fruit fly                                              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Equus caballus                  domestic horse, equine, horse                          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Felis catus                     cat, cats, domestic cat                                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gallus gallus                   bantam, chicken, chickens, Gallus domesticus           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Homo sapiens                    human                                                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Macaca mulatta                  rhesus macaque, rhesus macaques, Rhesus monkey, rhesus…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Monodelphis domestica           gray short-tailed opossum                              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> Mus musculus                    house mouse, mouse                                     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> Ornithorhynchus anatinus        duck-billed platypus, duckbill platypus, platypus      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> Pan troglodytes                 chimpanzee                                             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> Rattus norvegicus               brown rat, Norway rat, rat, rats                       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> Saccharomyces cerevisiae        baker's yeast, brewer's yeast, S. cerevisiae           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> Schizosaccharomyces pombe 972h- <span style="color: #BB0000;">NA</span>                                                     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> Sus scrofa                      pig, pigs, swine, wild boar                            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> Xenopus tropicalis              tropical clawed frog, western clawed frog</span></span></code></pre></div>
<p>You can retrieve data for a specific collection, such as the hallmark gene sets.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">h_gene_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 18</span></span></span>
<span><span class="co">#&gt;   gs_cat gs_subcat gs_name  gene_symbol entrez_gene ensembl_gene human_gene…¹ human_entr…²</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> H      <span style="color: #949494;">""</span>        HALLMAR… Abca1             <span style="text-decoration: underline;">11</span>303 ENSMUSG0000… ABCA1                  19</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> H      <span style="color: #949494;">""</span>        HALLMAR… Abcb8             <span style="text-decoration: underline;">74</span>610 ENSMUSG0000… ABCB8               <span style="text-decoration: underline;">11</span>194</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acaa2             <span style="text-decoration: underline;">52</span>538 ENSMUSG0000… ACAA2               <span style="text-decoration: underline;">10</span>449</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acadl             <span style="text-decoration: underline;">11</span>363 ENSMUSG0000… ACADL                  33</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acadm             <span style="text-decoration: underline;">11</span>364 ENSMUSG0000… ACADM                  34</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acads             <span style="text-decoration: underline;">11</span>409 ENSMUSG0000… ACADS                  35</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 10 more variables: human_ensembl_gene &lt;chr&gt;, gs_id &lt;chr&gt;, gs_pmid &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;, gs_description &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   taxon_id &lt;int&gt;, ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;, and abbreviated</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variable names ¹​human_gene_symbol, ²​human_entrez_gene</span></span></span></code></pre></div>
<p>Retrieve mouse C2 (curated) CGP (chemical and genetic perturbations) gene sets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cgp_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, category <span class="op">=</span> <span class="st">"C2"</span>, subcategory <span class="op">=</span> <span class="st">"CGP"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cgp_gene_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 18</span></span></span>
<span><span class="co">#&gt;   gs_cat gs_subcat gs_name  gene_symbol entrez_gene ensembl_gene human_gene…¹ human_entr…²</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> C2     CGP       ABBUD_L… Ahnak             <span style="text-decoration: underline;">66</span>395 ENSMUSG0000… AHNAK               <span style="text-decoration: underline;">79</span>026</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> C2     CGP       ABBUD_L… Alcam             <span style="text-decoration: underline;">11</span>658 ENSMUSG0000… ALCAM                 214</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> C2     CGP       ABBUD_L… Ankrd40           <span style="text-decoration: underline;">71</span>452 ENSMUSG0000… ANKRD40             <span style="text-decoration: underline;">91</span>369</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> C2     CGP       ABBUD_L… Arid1a            <span style="text-decoration: underline;">93</span>760 ENSMUSG0000… ARID1A               <span style="text-decoration: underline;">8</span>289</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> C2     CGP       ABBUD_L… Bckdhb            <span style="text-decoration: underline;">12</span>040 ENSMUSG0000… BCKDHB                594</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> C2     CGP       ABBUD_L… AU021092         <span style="text-decoration: underline;">239</span>691 ENSMUSG0000… C16orf89           <span style="text-decoration: underline;">146</span>556</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 10 more variables: human_ensembl_gene &lt;chr&gt;, gs_id &lt;chr&gt;, gs_pmid &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;, gs_description &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   taxon_id &lt;int&gt;, ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;, and abbreviated</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variable names ¹​human_gene_symbol, ²​human_entrez_gene</span></span></span></code></pre></div>
<p>There is a helper function to show the available collections.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/msigdbr_collections.html">msigdbr_collections</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 23 × 3</span></span></span>
<span><span class="co">#&gt;    gs_cat gs_subcat         num_genesets</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> C1     <span style="color: #949494;">""</span>                         299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> C2     <span style="color: #949494;">"</span>CGP<span style="color: #949494;">"</span>                     <span style="text-decoration: underline;">3</span>384</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> C2     <span style="color: #949494;">"</span>CP<span style="color: #949494;">"</span>                        29</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> C2     <span style="color: #949494;">"</span>CP:BIOCARTA<span style="color: #949494;">"</span>              292</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> C2     <span style="color: #949494;">"</span>CP:KEGG<span style="color: #949494;">"</span>                  186</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> C2     <span style="color: #949494;">"</span>CP:PID<span style="color: #949494;">"</span>                   196</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> C2     <span style="color: #949494;">"</span>CP:REACTOME<span style="color: #949494;">"</span>             <span style="text-decoration: underline;">1</span>615</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> C2     <span style="color: #949494;">"</span>CP:WIKIPATHWAYS<span style="color: #949494;">"</span>          664</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> C3     <span style="color: #949494;">"</span>MIR:MIR_Legacy<span style="color: #949494;">"</span>           221</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> C3     <span style="color: #949494;">"</span>MIR:MIRDB<span style="color: #949494;">"</span>               <span style="text-decoration: underline;">2</span>377</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> C3     <span style="color: #949494;">"</span>TFT:GTRD<span style="color: #949494;">"</span>                 518</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> C3     <span style="color: #949494;">"</span>TFT:TFT_Legacy<span style="color: #949494;">"</span>           610</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> C4     <span style="color: #949494;">"</span>CGN<span style="color: #949494;">"</span>                      427</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> C4     <span style="color: #949494;">"</span>CM<span style="color: #949494;">"</span>                       431</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> C5     <span style="color: #949494;">"</span>GO:BP<span style="color: #949494;">"</span>                   <span style="text-decoration: underline;">7</span>658</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> C5     <span style="color: #949494;">"</span>GO:CC<span style="color: #949494;">"</span>                   <span style="text-decoration: underline;">1</span>006</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> C5     <span style="color: #949494;">"</span>GO:MF<span style="color: #949494;">"</span>                   <span style="text-decoration: underline;">1</span>738</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> C5     <span style="color: #949494;">"</span>HPO<span style="color: #949494;">"</span>                     <span style="text-decoration: underline;">5</span>071</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> C6     <span style="color: #949494;">""</span>                         189</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> C7     <span style="color: #949494;">"</span>IMMUNESIGDB<span style="color: #949494;">"</span>             <span style="text-decoration: underline;">4</span>872</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">21</span> C7     <span style="color: #949494;">"</span>VAX<span style="color: #949494;">"</span>                      347</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">22</span> C8     <span style="color: #949494;">""</span>                         700</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">23</span> H      <span style="color: #949494;">""</span>                          50</span></span></code></pre></div>
<p>The <code><a href="../reference/msigdbr.html">msigdbr()</a></code> function output is a data frame and can be manipulated using more standard methods.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_gene_sets</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gs_cat</span> <span class="op">==</span> <span class="st">"H"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 18</span></span></span>
<span><span class="co">#&gt;   gs_cat gs_subcat gs_name  gene_symbol entrez_gene ensembl_gene human_gene…¹ human_entr…²</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> H      <span style="color: #949494;">""</span>        HALLMAR… Abca1             <span style="text-decoration: underline;">11</span>303 ENSMUSG0000… ABCA1                  19</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> H      <span style="color: #949494;">""</span>        HALLMAR… Abcb8             <span style="text-decoration: underline;">74</span>610 ENSMUSG0000… ABCB8               <span style="text-decoration: underline;">11</span>194</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acaa2             <span style="text-decoration: underline;">52</span>538 ENSMUSG0000… ACAA2               <span style="text-decoration: underline;">10</span>449</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acadl             <span style="text-decoration: underline;">11</span>363 ENSMUSG0000… ACADL                  33</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acadm             <span style="text-decoration: underline;">11</span>364 ENSMUSG0000… ACADM                  34</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> H      <span style="color: #949494;">""</span>        HALLMAR… Acads             <span style="text-decoration: underline;">11</span>409 ENSMUSG0000… ACADS                  35</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 10 more variables: human_ensembl_gene &lt;chr&gt;, gs_id &lt;chr&gt;, gs_pmid &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gs_geoid &lt;chr&gt;, gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;, gs_description &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   taxon_id &lt;int&gt;, ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;, and abbreviated</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variable names ¹​human_gene_symbol, ²​human_entrez_gene</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pathway-enrichment-analysis">Pathway enrichment analysis<a class="anchor" aria-label="anchor" href="#pathway-enrichment-analysis"></a>
</h2>
<p>The <code>msigdbr</code> output can be used with various popular pathway analysis packages.</p>
<p>Use the gene sets data frame for <code>clusterProfiler</code> with genes as Entrez Gene IDs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_t2g</span> <span class="op">=</span> <span class="va">msigdbr_df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">entrez_gene</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_ids_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <code>clusterProfiler</code> with genes as gene symbols.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_t2g</span> <span class="op">=</span> <span class="va">msigdbr_df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">gene_symbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_symbols_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <code>fgsea</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="fu">fgsea</span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <code>GSVA</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="fu">gsva</span><span class="op">(</span>gset.idx.list <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="potential-questions-or-concerns">Potential questions or concerns<a class="anchor" aria-label="anchor" href="#potential-questions-or-concerns"></a>
</h2>
<p><strong>Which version of MSigDB was used?</strong></p>
<p>This package was generated with MSigDB v7.5.1 (released January 2022). The MSigDB version is used as the base of the <code>msigdsbr</code> package version. You can check the installed version with <code>packageVersion("msigdbr")</code>.</p>
<p><strong>Can I download the gene sets directly from MSigDB instead of using this package?</strong></p>
<p>Yes. You can then import the GMT files (with <code>getGmt()</code> from the <code>GSEABase</code> package, for example). The GMTs only include the human genes, even for gene sets generated from mouse experiments. If you are working with non-human data, you then have to convert the MSigDB genes to your organism or your genes to human.</p>
<p><strong>Can I convert between human and mouse genes just by adjusting gene capitalization?</strong></p>
<p>That will work for most genes, but not all.</p>
<p><strong>Can I convert human genes to any organism myself instead of using this package?</strong></p>
<p>Yes. A popular method is using the <code>biomaRt</code> package. You may still end up with dozens of homologs for some genes, so additional cleanup may be helpful.</p>
<p><strong>Aren’t there already other similar tools?</strong></p>
<p>There are a few other resources that and provide some of the functionality and served as an inspiration for this package. <a href="http://ge-lab.org/#/data" class="external-link">Ge Lab Gene Set Files</a> has GMT files for many species. <a href="https://bioinf.wehi.edu.au/software/MSigDB/" class="external-link">WEHI</a> provides MSigDB gene sets in R format for human and mouse. <a href="https://github.com/stephenturner/msigdf" class="external-link">MSigDF</a> is based on the WEHI resource, but is converted to a more tidyverse-friendly data frame. These are updated at varying frequencies and may not use the latest version of MSigDB.</p>
<p><strong>What if I have other questions?</strong></p>
<p>You can submit feedback and report bugs on <a href="https://github.com/igordot/msigdbr/issues" class="external-link">GitHub</a>.</p>
</div>
<div class="section level2">
<h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a>
</h2>
<p>The Molecular Signatures Database (MSigDB) is a collection of gene sets originally created for use with the Gene Set Enrichment Analysis (GSEA) software. To cite use of the underlying MSigDB data, reference Subramanian, Tamayo, et al. (2005, PNAS) and one or more of the following as appropriate: Liberzon, et al. (2011, Bioinformatics), Liberzon, et al. (2015, Cell Systems), and also the source for the gene set.</p>
<p>Gene homologs are provided by HUGO Gene Nomenclature Committee at the European Bioinformatics Institute which integrates the orthology assertions predicted for human genes by eggNOG, Ensembl Compara, HGNC, HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL, Panther, PhylomeDB, TreeFam and ZFIN. For each human equivalent within each species, only the ortholog supported by the largest number of databases is used.</p>
<p>For information on how to cite cite an R package such as <code>msigdbr</code>, you can execute <code>citation("msigdbr")</code>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Igor Dolgalev.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
