<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to msigdbr • msigdbr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to msigdbr">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">msigdbr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">10.0.2.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<ul class="navbar-nav">
<li class="active nav-item"><a class="nav-link" href="../articles/msigdbr-intro.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/igordot/msigdbr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="msigdbr-intro_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="msigdbr-intro_files/pagedtable-1.1/js/pagedtable.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to msigdbr</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/igordot/msigdbr/blob/main/vignettes/articles/msigdbr-intro.Rmd" class="external-link"><code>vignettes/articles/msigdbr-intro.Rmd</code></a></small>
      <div class="d-none name"><code>msigdbr-intro.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Pathway analysis is a common task in genomics research and there are
many available R-based software tools. Depending on the tool, it may be
necessary to import the pathways, translate genes to the appropriate
species, convert between symbols and IDs, and format the resulting
object.</p>
<p>The msigdbr R package provides Molecular Signatures Database (MSigDB)
gene sets typically used with the Gene Set Enrichment Analysis (GSEA)
software:</p>
<ul>
<li>in an R-friendly “<a href="https://r4ds.hadley.nz/data-tidy.html" class="external-link">tidy</a>” format with one
gene pair per row</li>
<li>for multiple frequently studied model organisms, such as mouse, rat,
pig, zebrafish, fly, and yeast, in addition to the original human
genes</li>
<li>as gene symbols as well as NCBI Entrez and Ensembl IDs</li>
<li>without accessing external resources requiring an active internet
connection</li>
</ul>
<p>Please be aware that the orthologs were computationally predicted at
the gene level. The full pathways may not be well conserved across
species.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The package can be installed from <a href="https://cran.r-project.org/package=msigdbr" class="external-link">CRAN</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"msigdbr"</span><span class="op">)</span></span></code></pre></div>
<p>The package includes only a small subset of the full MSigDB database
due to CRAN size limitations. Please install the msigdbdf package to
access the full MSigDB database:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"msigdbdf"</span>, repos <span class="op">=</span> <span class="st">"https://igordot.r-universe.dev"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Load package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/">msigdbr</a></span><span class="op">)</span></span></code></pre></div>
<p>All gene sets in the database can be retrieved by specifying a
species of interest.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">all_gene_sets</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["gene_symbol"],"name":[1],"type":["chr"],"align":["left"]},{"label":["ncbi_gene"],"name":[2],"type":["chr"],"align":["left"]},{"label":["ensembl_gene"],"name":[3],"type":["chr"],"align":["left"]},{"label":["db_gene_symbol"],"name":[4],"type":["chr"],"align":["left"]},{"label":["db_ncbi_gene"],"name":[5],"type":["chr"],"align":["left"]},{"label":["db_ensembl_gene"],"name":[6],"type":["chr"],"align":["left"]},{"label":["source_gene"],"name":[7],"type":["chr"],"align":["left"]},{"label":["gs_id"],"name":[8],"type":["chr"],"align":["left"]},{"label":["gs_name"],"name":[9],"type":["chr"],"align":["left"]},{"label":["gs_collection"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gs_subcollection"],"name":[11],"type":["chr"],"align":["left"]},{"label":["gs_collection_name"],"name":[12],"type":["chr"],"align":["left"]},{"label":["gs_description"],"name":[13],"type":["chr"],"align":["left"]},{"label":["gs_source_species"],"name":[14],"type":["chr"],"align":["left"]},{"label":["gs_pmid"],"name":[15],"type":["chr"],"align":["left"]},{"label":["gs_geoid"],"name":[16],"type":["chr"],"align":["left"]},{"label":["gs_exact_source"],"name":[17],"type":["chr"],"align":["left"]},{"label":["gs_url"],"name":[18],"type":["chr"],"align":["left"]},{"label":["db_version"],"name":[19],"type":["chr"],"align":["left"]},{"label":["db_target_species"],"name":[20],"type":["chr"],"align":["left"]},{"label":["ortholog_taxon_id"],"name":[21],"type":["int"],"align":["right"]},{"label":["ortholog_sources"],"name":[22],"type":["chr"],"align":["left"]},{"label":["num_ortholog_sources"],"name":[23],"type":["dbl"],"align":["right"]}],"data":[{"1":"Abcc4","2":"239273","3":"ENSMUSG00000032849","4":"ABCC4","5":"10257","6":"ENSG00000125257","7":"ABCC4","8":"M12609","9":"AAACCAC_MIR140","10":"C3","11":"MIR:MIR_LEGACY","12":"MIR_Legacy","13":"Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).","14":"HS","15":"","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther","23":"9"},{"1":"Abraxas2","2":"109359","3":"ENSMUSG00000030965","4":"ABRAXAS2","5":"23172","6":"ENSG00000165660","7":"KIAA0157","8":"M12609","9":"AAACCAC_MIR140","10":"C3","11":"MIR:MIR_LEGACY","12":"MIR_Legacy","13":"Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).","14":"HS","15":"","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Actn4","2":"60595","3":"ENSMUSG00000054808","4":"ACTN4","5":"81","6":"ENSG00000130402","7":"ACTN4","8":"M12609","9":"AAACCAC_MIR140","10":"C3","11":"MIR:MIR_LEGACY","12":"MIR_Legacy","13":"Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).","14":"HS","15":"","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Acvr1","2":"11477","3":"ENSMUSG00000026836","4":"ACVR1","5":"90","6":"ENSG00000115170","7":"ACVR1","8":"M12609","9":"AAACCAC_MIR140","10":"C3","11":"MIR:MIR_LEGACY","12":"MIR_Legacy","13":"Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).","14":"HS","15":"","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Adam9","2":"11502","3":"ENSMUSG00000031555","4":"ADAM9","5":"8754","6":"ENSG00000168615","7":"ADAM9","8":"M12609","9":"AAACCAC_MIR140","10":"C3","11":"MIR:MIR_LEGACY","12":"MIR_Legacy","13":"Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).","14":"HS","15":"","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"},{"1":"Adamts5","2":"23794","3":"ENSMUSG00000022894","4":"ADAMTS5","5":"11096","6":"ENSG00000154736","7":"ADAMTS5","8":"M12609","9":"AAACCAC_MIR140","10":"C3","11":"MIR:MIR_LEGACY","12":"MIR_Legacy","13":"Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).","14":"HS","15":"","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[30],"max":[30]},"pages":{}}}
  </script>
</div>
<p>The <code>gene_symbol</code>, <code>ncbi_gene</code>, and
<code>ensembl_gene</code> columns are the gene identifiers for the
requested species. The <code>db_gene_symbol</code>,
<code>db_ncbi_gene</code>, <code>db_ensembl_gene</code> columns are the
gene identifiers in the database. The <code>source_gene</code> column
are the gene identifiers in the original publication. The
<code>gs_*</code> columns provide details about the gene sets. The
<code>db_*</code> columns provide details about the MSigDB database, so
should be identical for the entire data frame.</p>
<p>You can retrieve data just for a specific collection, such as the
Hallmark gene sets.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, collection <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">h_gene_sets</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["gene_symbol"],"name":[1],"type":["chr"],"align":["left"]},{"label":["ncbi_gene"],"name":[2],"type":["chr"],"align":["left"]},{"label":["ensembl_gene"],"name":[3],"type":["chr"],"align":["left"]},{"label":["db_gene_symbol"],"name":[4],"type":["chr"],"align":["left"]},{"label":["db_ncbi_gene"],"name":[5],"type":["chr"],"align":["left"]},{"label":["db_ensembl_gene"],"name":[6],"type":["chr"],"align":["left"]},{"label":["source_gene"],"name":[7],"type":["chr"],"align":["left"]},{"label":["gs_id"],"name":[8],"type":["chr"],"align":["left"]},{"label":["gs_name"],"name":[9],"type":["chr"],"align":["left"]},{"label":["gs_collection"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gs_subcollection"],"name":[11],"type":["chr"],"align":["left"]},{"label":["gs_collection_name"],"name":[12],"type":["chr"],"align":["left"]},{"label":["gs_description"],"name":[13],"type":["chr"],"align":["left"]},{"label":["gs_source_species"],"name":[14],"type":["chr"],"align":["left"]},{"label":["gs_pmid"],"name":[15],"type":["chr"],"align":["left"]},{"label":["gs_geoid"],"name":[16],"type":["chr"],"align":["left"]},{"label":["gs_exact_source"],"name":[17],"type":["chr"],"align":["left"]},{"label":["gs_url"],"name":[18],"type":["chr"],"align":["left"]},{"label":["db_version"],"name":[19],"type":["chr"],"align":["left"]},{"label":["db_target_species"],"name":[20],"type":["chr"],"align":["left"]},{"label":["ortholog_taxon_id"],"name":[21],"type":["int"],"align":["right"]},{"label":["ortholog_sources"],"name":[22],"type":["chr"],"align":["left"]},{"label":["num_ortholog_sources"],"name":[23],"type":["dbl"],"align":["right"]}],"data":[{"1":"Abca1","2":"11303","3":"ENSMUSG00000015243","4":"ABCA1","5":"19","6":"ENSG00000165029","7":"ABCA1","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"},{"1":"Abcb8","2":"74610","3":"ENSMUSG00000028973","4":"ABCB8","5":"11194","6":"ENSG00000197150","7":"ABCB8","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Acaa2","2":"52538","3":"ENSMUSG00000036880","4":"ACAA2","5":"10449","6":"ENSG00000167315","7":"ACAA2","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"},{"1":"Acadl","2":"11363","3":"ENSMUSG00000026003","4":"ACADL","5":"33","6":"ENSG00000115361","7":"ACADL","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Acadm","2":"11364","3":"ENSMUSG00000062908","4":"ACADM","5":"34","6":"ENSG00000117054","7":"ACADM","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Acads","2":"11409","3":"ENSMUSG00000029545","4":"ACADS","5":"35","6":"ENSG00000122971","7":"ACADS","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[30],"max":[30]},"pages":{}}}
  </script>
</div>
<p>You can specify a sub-collection, such as C2 (curated) CGP (chemical
and genetic perturbations) gene sets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cgp_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, collection <span class="op">=</span> <span class="st">"C2"</span>, subcollection <span class="op">=</span> <span class="st">"CGP"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cgp_gene_sets</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["gene_symbol"],"name":[1],"type":["chr"],"align":["left"]},{"label":["ncbi_gene"],"name":[2],"type":["chr"],"align":["left"]},{"label":["ensembl_gene"],"name":[3],"type":["chr"],"align":["left"]},{"label":["db_gene_symbol"],"name":[4],"type":["chr"],"align":["left"]},{"label":["db_ncbi_gene"],"name":[5],"type":["chr"],"align":["left"]},{"label":["db_ensembl_gene"],"name":[6],"type":["chr"],"align":["left"]},{"label":["source_gene"],"name":[7],"type":["chr"],"align":["left"]},{"label":["gs_id"],"name":[8],"type":["chr"],"align":["left"]},{"label":["gs_name"],"name":[9],"type":["chr"],"align":["left"]},{"label":["gs_collection"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gs_subcollection"],"name":[11],"type":["chr"],"align":["left"]},{"label":["gs_collection_name"],"name":[12],"type":["chr"],"align":["left"]},{"label":["gs_description"],"name":[13],"type":["chr"],"align":["left"]},{"label":["gs_source_species"],"name":[14],"type":["chr"],"align":["left"]},{"label":["gs_pmid"],"name":[15],"type":["chr"],"align":["left"]},{"label":["gs_geoid"],"name":[16],"type":["chr"],"align":["left"]},{"label":["gs_exact_source"],"name":[17],"type":["chr"],"align":["left"]},{"label":["gs_url"],"name":[18],"type":["chr"],"align":["left"]},{"label":["db_version"],"name":[19],"type":["chr"],"align":["left"]},{"label":["db_target_species"],"name":[20],"type":["chr"],"align":["left"]},{"label":["ortholog_taxon_id"],"name":[21],"type":["int"],"align":["right"]},{"label":["ortholog_sources"],"name":[22],"type":["chr"],"align":["left"]},{"label":["num_ortholog_sources"],"name":[23],"type":["dbl"],"align":["right"]}],"data":[{"1":"Ahnak","2":"66395","3":"ENSMUSG00000069833","4":"AHNAK","5":"79026","6":"ENSG00000124942","7":"AA657044","8":"M1423","9":"ABBUD_LIF_SIGNALING_1_DN","10":"C2","11":"CGP","12":"Chemical and Genetic Perturbations","13":"Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].","14":"MM","15":"14576184","16":"","17":"Table 2","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"},{"1":"Alcam","2":"11658","3":"ENSMUSG00000022636","4":"ALCAM","5":"214","6":"ENSG00000170017","7":"L25274","8":"M1423","9":"ABBUD_LIF_SIGNALING_1_DN","10":"C2","11":"CGP","12":"Chemical and Genetic Perturbations","13":"Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].","14":"MM","15":"14576184","16":"","17":"Table 2","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Ankrd40","2":"71452","3":"ENSMUSG00000020864","4":"ANKRD40","5":"91369","6":"ENSG00000154945","7":"AW125433","8":"M1423","9":"ABBUD_LIF_SIGNALING_1_DN","10":"C2","11":"CGP","12":"Chemical and Genetic Perturbations","13":"Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].","14":"MM","15":"14576184","16":"","17":"Table 2","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Arid1a","2":"93760","3":"ENSMUSG00000007880","4":"ARID1A","5":"8289","6":"ENSG00000117713","7":"AI842326","8":"M1423","9":"ABBUD_LIF_SIGNALING_1_DN","10":"C2","11":"CGP","12":"Chemical and Genetic Perturbations","13":"Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].","14":"MM","15":"14576184","16":"","17":"Table 2","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Bckdhb","2":"12040","3":"ENSMUSG00000032263","4":"BCKDHB","5":"594","6":"ENSG00000083123","7":"L16992","8":"M1423","9":"ABBUD_LIF_SIGNALING_1_DN","10":"C2","11":"CGP","12":"Chemical and Genetic Perturbations","13":"Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].","14":"MM","15":"14576184","16":"","17":"Table 2","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"AU021092","2":"239691","3":"ENSMUSG00000051669","4":"C16orf89","5":"146556","6":"ENSG00000153446","7":"AW045808","8":"M1423","9":"ABBUD_LIF_SIGNALING_1_DN","10":"C2","11":"CGP","12":"Chemical and Genetic Perturbations","13":"Genes down-regulated in AtT20 cells (pituitary cancer) after  treatment with LIF [GeneID=3976].","14":"MM","15":"14576184","16":"","17":"Table 2","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[30],"max":[30]},"pages":{}}}
  </script>
</div>
<p>If you require more precise filtering, the <code><a href="../reference/msigdbr.html">msigdbr()</a></code>
function output is a data frame that can be manipulated using standard
methods. For example, you can subset to a specific collection using
dplyr.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_gene_sets</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gs_collection</span> <span class="op">==</span> <span class="st">"H"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["gene_symbol"],"name":[1],"type":["chr"],"align":["left"]},{"label":["ncbi_gene"],"name":[2],"type":["chr"],"align":["left"]},{"label":["ensembl_gene"],"name":[3],"type":["chr"],"align":["left"]},{"label":["db_gene_symbol"],"name":[4],"type":["chr"],"align":["left"]},{"label":["db_ncbi_gene"],"name":[5],"type":["chr"],"align":["left"]},{"label":["db_ensembl_gene"],"name":[6],"type":["chr"],"align":["left"]},{"label":["source_gene"],"name":[7],"type":["chr"],"align":["left"]},{"label":["gs_id"],"name":[8],"type":["chr"],"align":["left"]},{"label":["gs_name"],"name":[9],"type":["chr"],"align":["left"]},{"label":["gs_collection"],"name":[10],"type":["chr"],"align":["left"]},{"label":["gs_subcollection"],"name":[11],"type":["chr"],"align":["left"]},{"label":["gs_collection_name"],"name":[12],"type":["chr"],"align":["left"]},{"label":["gs_description"],"name":[13],"type":["chr"],"align":["left"]},{"label":["gs_source_species"],"name":[14],"type":["chr"],"align":["left"]},{"label":["gs_pmid"],"name":[15],"type":["chr"],"align":["left"]},{"label":["gs_geoid"],"name":[16],"type":["chr"],"align":["left"]},{"label":["gs_exact_source"],"name":[17],"type":["chr"],"align":["left"]},{"label":["gs_url"],"name":[18],"type":["chr"],"align":["left"]},{"label":["db_version"],"name":[19],"type":["chr"],"align":["left"]},{"label":["db_target_species"],"name":[20],"type":["chr"],"align":["left"]},{"label":["ortholog_taxon_id"],"name":[21],"type":["int"],"align":["right"]},{"label":["ortholog_sources"],"name":[22],"type":["chr"],"align":["left"]},{"label":["num_ortholog_sources"],"name":[23],"type":["dbl"],"align":["right"]}],"data":[{"1":"Abca1","2":"11303","3":"ENSMUSG00000015243","4":"ABCA1","5":"19","6":"ENSG00000165029","7":"ABCA1","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"},{"1":"Abcb8","2":"74610","3":"ENSMUSG00000028973","4":"ABCB8","5":"11194","6":"ENSG00000197150","7":"ABCB8","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Acaa2","2":"52538","3":"ENSMUSG00000036880","4":"ACAA2","5":"10449","6":"ENSG00000167315","7":"ACAA2","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"},{"1":"Acadl","2":"11363","3":"ENSMUSG00000026003","4":"ACADL","5":"33","6":"ENSG00000115361","7":"ACADL","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Acadm","2":"11364","3":"ENSMUSG00000062908","4":"ACADM","5":"34","6":"ENSG00000117054","7":"ACADM","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam","23":"12"},{"1":"Acads","2":"11409","3":"ENSMUSG00000029545","4":"ACADS","5":"35","6":"ENSG00000122971","7":"ACADS","8":"M5905","9":"HALLMARK_ADIPOGENESIS","10":"H","11":"","12":"Hallmark","13":"Genes up-regulated during adipocyte differentiation (adipogenesis).","14":"HS","15":"26771021","16":"","17":"","18":"","19":"2024.1.Hs","20":"HS","21":"10090","22":"EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam","23":"11"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[30],"max":[30]},"pages":{}}}
  </script>
</div>
<p>The version of the MSigDB database is stored in the
<code>db_version</code> column of the returned data frame.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">all_gene_sets</span><span class="op">$</span><span class="va">db_version</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "2024.1.Hs"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="helper-functions">Helper functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a>
</h2>
<p>There are helper functions to assist with setting the
<code><a href="../reference/msigdbr.html">msigdbr()</a></code> parameters.</p>
<p>Use <code><a href="../reference/msigdbr_species.html">msigdbr_species()</a></code> to check the available species.
Both scientific and common names are acceptable for the
<code><a href="../reference/msigdbr.html">msigdbr()</a></code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/msigdbr_species.html">msigdbr_species</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["species_name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["species_common_name"],"name":[2],"type":["chr"],"align":["left"]}],"data":[{"1":"Anolis carolinensis","2":"Carolina anole, green anole"},{"1":"Bos taurus","2":"bovine, cattle, cow, dairy cow, domestic cattle, domestic cow, ox, oxen"},{"1":"Caenorhabditis elegans","2":"NA"},{"1":"Canis lupus familiaris","2":"dog, dogs"},{"1":"Danio rerio","2":"leopard danio, zebra danio, zebra fish, zebrafish"},{"1":"Drosophila melanogaster","2":"fruit fly"},{"1":"Equus caballus","2":"domestic horse, equine, horse"},{"1":"Felis catus","2":"cat, cats, domestic cat"},{"1":"Gallus gallus","2":"bantam, chicken, chickens, Gallus domesticus"},{"1":"Homo sapiens","2":"human"},{"1":"Macaca mulatta","2":"rhesus macaque, rhesus macaques, Rhesus monkey, rhesus monkeys"},{"1":"Monodelphis domestica","2":"gray short-tailed opossum"},{"1":"Mus musculus","2":"house mouse, mouse"},{"1":"Ornithorhynchus anatinus","2":"duck-billed platypus, duckbill platypus, platypus"},{"1":"Pan troglodytes","2":"chimpanzee"},{"1":"Rattus norvegicus","2":"brown rat, Norway rat, rat, rats"},{"1":"Saccharomyces cerevisiae","2":"baker's yeast, brewer's yeast, S. cerevisiae"},{"1":"Schizosaccharomyces pombe 972h-","2":"NA"},{"1":"Sus scrofa","2":"pig, pigs, swine, wild boar"},{"1":"Xenopus tropicalis","2":"tropical clawed frog, western clawed frog"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[30],"max":[30]},"pages":{}}}
  </script>
</div>
<p>Use <code><a href="../reference/msigdbr_collections.html">msigdbr_collections()</a></code> to check the available
collections.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/msigdbr_collections.html">msigdbr_collections</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["gs_collection"],"name":[1],"type":["chr"],"align":["left"]},{"label":["gs_subcollection"],"name":[2],"type":["chr"],"align":["left"]},{"label":["gs_collection_name"],"name":[3],"type":["chr"],"align":["left"]},{"label":["num_genesets"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"C1","2":"","3":"Positional","4":"302"},{"1":"C2","2":"CGP","3":"Chemical and Genetic Perturbations","4":"3494"},{"1":"C2","2":"CP","3":"Canonical Pathways","4":"19"},{"1":"C2","2":"CP:BIOCARTA","3":"BioCarta Pathways","4":"292"},{"1":"C2","2":"CP:KEGG_LEGACY","3":"KEGG Legacy Pathways","4":"186"},{"1":"C2","2":"CP:KEGG_MEDICUS","3":"KEGG Medicus Pathways","4":"658"},{"1":"C2","2":"CP:PID","3":"PID Pathways","4":"196"},{"1":"C2","2":"CP:REACTOME","3":"Reactome Pathways","4":"1736"},{"1":"C2","2":"CP:WIKIPATHWAYS","3":"WikiPathways","4":"830"},{"1":"C3","2":"MIR:MIRDB","3":"miRDB","4":"2377"},{"1":"C3","2":"MIR:MIR_LEGACY","3":"MIR_Legacy","4":"221"},{"1":"C3","2":"TFT:GTRD","3":"GTRD","4":"505"},{"1":"C3","2":"TFT:TFT_LEGACY","3":"TFT_Legacy","4":"610"},{"1":"C4","2":"3CA","3":"Curated Cancer Cell Atlas gene sets","4":"148"},{"1":"C4","2":"CGN","3":"Cancer Gene Neighborhoods","4":"427"},{"1":"C4","2":"CM","3":"Cancer Modules","4":"431"},{"1":"C5","2":"GO:BP","3":"GO Biological Process","4":"7608"},{"1":"C5","2":"GO:CC","3":"GO Cellular Component","4":"1026"},{"1":"C5","2":"GO:MF","3":"GO Molecular Function","4":"1820"},{"1":"C5","2":"HPO","3":"Human Phenotype Ontology","4":"5653"},{"1":"C6","2":"","3":"Oncogenic Signature","4":"189"},{"1":"C7","2":"IMMUNESIGDB","3":"ImmuneSigDB","4":"4872"},{"1":"C7","2":"VAX","3":"HIPC Vaccine Response","4":"347"},{"1":"C8","2":"","3":"Cell Type Signature","4":"840"},{"1":"H","2":"","3":"Hallmark","4":"50"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[30],"max":[30]},"pages":{}}}
  </script>
</div>
</div>
<div class="section level2">
<h2 id="pathway-enrichment-analysis">Pathway enrichment analysis<a class="anchor" aria-label="anchor" href="#pathway-enrichment-analysis"></a>
</h2>
<p>The msigdbr output can be used with various pathway analysis
packages.</p>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/clusterProfiler/" class="external-link">clusterProfiler</a>
with genes as NCBI/Entrez IDs.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_t2g</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">msigdbr_df</span>, <span class="va">gs_name</span>, <span class="va">ncbi_gene</span><span class="op">)</span></span>
<span><span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_ids_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/clusterProfiler/" class="external-link">clusterProfiler</a>
with genes as gene symbols.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_t2g</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">msigdbr_df</span>, <span class="va">gs_name</span>, <span class="va">gene_symbol</span><span class="op">)</span></span>
<span><span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_symbols_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/fgsea/" class="external-link">fgsea</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="fu">fgsea</span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>Use the gene sets data frame for <a href="https://bioconductor.org/packages/GSVA/" class="external-link">GSVA</a>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="va">gsvapar</span> <span class="op">&lt;-</span> <span class="fu">gsvaParam</span><span class="op">(</span>geneSets <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></span>
<span><span class="fu">gsva</span><span class="op">(</span><span class="va">gsvapar</span><span class="op">)</span></span></code></pre></div>
<p>Earlier versions of GSVA (&lt;1.50) only need the <code>gsva()</code>
function.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msigdbr_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="fu">gsva</span><span class="op">(</span>gset.idx.list <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="potential-questions-and-concerns">Potential questions and concerns<a class="anchor" aria-label="anchor" href="#potential-questions-and-concerns"></a>
</h2>
<p><strong>Which version of MSigDB was used?</strong></p>
<p>The MSigDB version is stored in the <code>db_version</code> column of
the returned data frame. You can check the version used with
<code>unique(msigdbr_df$db_version)</code>.</p>
<p><strong>Why use this package when I can download the gene sets
directly from MSigDB?</strong></p>
<p>This package makes it more convenient to work with MSigDB gene sets
in R. You don’t need to download the GMT files and import them. You
don’t need to learn how to restructure the output to make it compatible
with downstream tools. You don’t need to convert the genes to your
organism if you are working with non-human data.</p>
<p><strong>Can I convert between human and mouse genes just by adjusting
gene capitalization?</strong></p>
<p>That will work for most, but not all, genes.</p>
<p><strong>Can I convert human genes to any organism myself instead of
using this package?</strong></p>
<p>One popular method is using the biomaRt package. You may still end up
with dozens of homologs for some genes, so additional cleanup may be
helpful.</p>
<p><strong>Aren’t there already other similar tools?</strong></p>
<p>There are a few resources that provide some of the msigdbr
functionality and served as an inspiration for this package. <a href="https://bioinf.wehi.edu.au/software/MSigDB/" class="external-link">WEHI</a> provides
MSigDB gene sets in R format for human and mouse. <a href="https://github.com/stephenturner/msigdf" class="external-link">MSigDF</a> and a more
recent <a href="https://github.com/ToledoEM/msigdf" class="external-link">ToledoEM/msigdf</a>
fork provide a tidyverse-friendly data frame. These are updated at
varying frequencies and may not be based on the latest version of
MSigDB. Since 2022, the GSEA/MSigDB team provides <a href="https://www.gsea-msigdb.org/gsea/msigdb/mouse/collections.jsp" class="external-link">collections
that are natively mouse</a> and don’t require orthology conversion.</p>
<p><strong>What if I have other questions?</strong></p>
<p>You can submit feedback and report bugs on <a href="https://github.com/igordot/msigdbr/issues" class="external-link">GitHub</a>.</p>
</div>
<div class="section level2">
<h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a>
</h2>
<p>The Molecular Signatures Database (MSigDB) is a collection of gene
sets originally created for use with the Gene Set Enrichment Analysis
(GSEA) software. To cite use of the underlying MSigDB data, reference
Subramanian, Tamayo, et al. (2005, PNAS) and one or more of the
following as appropriate: Liberzon, et al. (2011, Bioinformatics),
Liberzon, et al. (2015, Cell Systems), Castanza, et al. (2023, Nature
Methods) and also the source for the gene set.</p>
<p>Gene homologs are provided by HUGO Gene Nomenclature Committee at the
European Bioinformatics Institute which integrates the orthology
assertions predicted for human genes by eggNOG, Ensembl Compara, HGNC,
HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL,
Panther, PhylomeDB, TreeFam and ZFIN. For each human equivalent within
each species, only the ortholog supported by the largest number of
databases is used.</p>
<p>For information on how to cite the msigdbr R package, run
<code>citation("msigdbr")</code>.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Igor Dolgalev.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>



  <script data-goatcounter="https://igordot.goatcounter.com/count" async src="//gc.zgo.at/count.js"></script>
</body>
</html>
